### PR TITLE
Feat/delete event

### LINKED ISSUE #60: Delete an event
User story: As an event organizer, I want to delete my event, so that it no longer appears if it is canceled or irrelevant.

### LINKED ISSUE #170: (1) Event Management - Edit & Delete
Allow event organizers to edit and delete their own events. Only the event creator (where event.ownerId == currentUserId) can perform these actions.

  Acceptance Criteria:
  - "Edit" and "Delete" buttons appear in event detail sheet ONLY for event owner
  - Edit button opens EditEventScreen with pre-filled event data
  - Can modify: title, description, date, time, location, tags, public/private, capacity
  - Cannot change: ownerId, uid, participantIds
  - Delete shows confirmation dialog before removing event
  - Deletion removes event from Firestore and all users' saved events
  -  Add EventStatus enum (DRAFT, PUBLISHED, CANCELLED, COMPLETED) to Event model
  -  Organizers can mark events as CANCELLED instead of deleting (preserves history)
  -  Show "Updated" badge on event if lastEditedAt: Timestamp? is set
  - Success/error feedback and proper validation
  - Tests for owner-only access and edge cases

### PR DESCRIPTION
**Pull request by ItsaBox9368**
## Description

This introduces the functionality to delete an owned event, it allows for better event flow organisation.

**Related Issue:**
Closes #60
Closes #170 

## Changes
Added the logic for deleting an event in the search results items DropDownMenu

**Implementation:**
- Added an alert dialog popup when a user selects delete event in the DropDownMenu for owned search result items
- Implemented state management functions in MapScreenViewModel
- Added a callback called onDeleteEvent in BottomSheetContent and MapaScreen to connect the existing logic from SearchResultItems

**Tests:**

- Add new Tests in MapScreenTests to test AlertDialog section
- Add new Tests in MapScreenViewModel to test delete event functions

## Testing
- [X] Unit tests pass
- [X] Instrumentation tests pass
- [X] Manual testing completed
- [X] Coverage maintained (≥80%)

**Test summary:**
Add 3 UI tests in MapScreenTests and 4 unit tests in MapScrenViewModel.

## Screenshots/Videos
https://github.com/user-attachments/assets/55002f68-2f5e-4a5a-b0bf-d016e7f46624

## Checklist
- [X] Sufficient documentation and minimal inline comments
- [X] All tests passing
- [X] No new warnings
- [X] If related issue exists: issue has all labels, fields, and milestone filled

### CODE REVIEW

File: app/src/main/java/com/swent/mapin/ui/map/MapScreen.kt [hunk 1]
```diff
@@ -679,6 +683,29 @@ fun MapScreen(
...
+                  }) {
+                    Text("Delete", color = Color.Red)
+                  }
+            },
+            dismissButton = {
+              TextButton(onClick = { viewModel.cancelDelete() }) { Text("Cancel") }
+            })
+      }
+    }
+
```
Comment (ziyad-elarroussi) [inline app/src/main/java/com/swent/mapin/ui/map/MapScreen.kt]: The nested let + if is redundant. 

If eventPendingDeletion is set, showDeleteDialog should always be  true (they're set together in requestDeleteEvent). This double-check  suggests the state management could be cleaner.

Pick one:
  - Just check showDeleteDialog and use eventPendingDeletion!!  inside ( I recommend this)

```kotlin
  if (viewModel.showDeleteDialog) {
    val eventToDelete = viewModel.eventPendingDeletion!!
    AlertDialog(...)
  }

```

  - Combine into single state object (as I mentioned earlier)

```kotlin
  viewModel.eventPendingDeletion?.let { eventToDelete ->
    AlertDialog(...)  // Remove the inner if
  }

```

File: app/src/main/java/com/swent/mapin/ui/map/MapScreen.kt [hunk 2]
```diff
@@ -679,6 +683,29 @@ fun MapScreen(
...
+        AlertDialog(
+            onDismissRequest = { viewModel.cancelDelete() },
+            title = { Text("Delete Event") },
+            text = {
+              Text("Are you sure you want to delete this event? This action cannot be undone.")
+            },
+            confirmButton = {
+              TextButton(
+                  onClick = {
+                    eventViewModel.deleteEvent(eventToDelete.uid)
```
[the remark this answers is not in the document]
Comment (ItsaBox9368) [reply app/src/main/java/com/swent/mapin/ui/map/MapScreen.kt]: Thanks for the advice, but I don't see what the problem is with calling a viewModel function in UI nor what business logic is.
There is also no viewModel.confirmDelete() so I don't know where else I would call it, it would be a viewModel function anyway.

File: app/src/main/java/com/swent/mapin/ui/map/MapScreen.kt [hunk 3]
```diff
@@ -679,6 +683,29 @@ fun MapScreen(
...
+                    viewModel.cancelDelete()
+                  }) {
+                    Text("Delete", color = Color.Red)
```
[the remark this answers is not in the document]
Comment (ItsaBox9368) [reply app/src/main/java/com/swent/mapin/ui/map/MapScreen.kt]: Wouldn't an error also be Red in dark mode? But sure thanks, changed.

File: app/src/main/java/com/swent/mapin/ui/map/MapScreen.kt [hunk 4]
```diff
@@ -679,6 +683,29 @@ fun MapScreen(
           event = viewModel.selectedEvent!!, onDismiss = { viewModel.dismissShareDialog() })
     }
 
+    viewModel.eventPendingDeletion?.let { eventToDelete ->
+      if (viewModel.showDeleteDialog) {
+        AlertDialog(
+            onDismissRequest = { viewModel.cancelDelete() },
+            title = { Text("Delete Event") },
+            text = {
+              Text("Are you sure you want to delete this event? This action cannot be undone.")
```
[the remark this answers is not in the document]
Comment (ItsaBox9368) [reply app/src/main/java/com/swent/mapin/ui/map/MapScreen.kt]: Thanks, it is now done.

**Review by ziyad-elarroussi — APPROVED** [review]
